src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/OtlpRetry.cs

```diff
@@ -0,0 +1,248 @@
+        retryResult = new RetryResult(throttleDelay.HasValue, delayDuration, nextRetryDelayMilliseconds);
+        return true;
+    }
+
+    private static bool IsDeadlineExceeded(DateTime? deadline)
+    {
+        // This implementation is internal, and it is guaranteed that deadline is UTC.
+        return deadline.HasValue && deadline <= DateTime.UtcNow;
+    }
+
+    private static int CalculateNextRetryDelay(int nextRetryDelayMilliseconds)
+    {
+        var nextMilliseconds = nextRetryDelayMilliseconds * BackoffMultiplier;
+        nextMilliseconds = Math.Min(nextMilliseconds, MaxBackoffMilliseconds);
+        return Convert.ToInt32(nextMilliseconds);
+    }
+
+    private static TimeSpan? TryGetGrpcRetryDelay(StatusCode statusCode, Metadata trailers)
+    {
+        Debug.Assert(trailers != null, "trailers was null");
+
+        if (statusCode != StatusCode.ResourceExhausted && statusCode != StatusCode.Unavailable)
+        {
+            return null;
+        }
+
+        var statusDetails = trailers.Get(GrpcStatusDetailsHeader);
+        if (statusDetails != null && statusDetails.IsBinary)
+        {
+            var status = Status.Parser.ParseFrom(statusDetails.ValueBytes);
+            foreach (var item in status.Details)
+            {
+                var success = item.TryUnpack<RetryInfo>(out var retryInfo);
+                if (success)
+                {
+                    return retryInfo.RetryDelay.ToTimeSpan();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private static TimeSpan? TryGetHttpRetryDelay(HttpStatusCode statusCode, HttpResponseHeaders headers)
+    {
+        Debug.Assert(headers != null, "headers was null");
+
+#if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
+        return statusCode == HttpStatusCode.TooManyRequests || statusCode == HttpStatusCode.ServiceUnavailable
+#else
+        return statusCode == (HttpStatusCode)429 || statusCode == HttpStatusCode.ServiceUnavailable
+#endif
+            ? headers.RetryAfter?.Delta
+            : null;
+    }
+
+    private static bool IsGrpcStatusCodeRetryable(StatusCode statusCode, bool hasRetryDelay)
+    {
+        switch (statusCode)
+        {
+            case StatusCode.Cancelled:
+            case StatusCode.DeadlineExceeded:
+            case StatusCode.Aborted:
+            case StatusCode.OutOfRange:
+            case StatusCode.Unavailable:
+            case StatusCode.DataLoss:
+                return true;
+            case StatusCode.ResourceExhausted:
+                return hasRetryDelay;
+            default:
+                return false;
+        }
+    }
+
+    private static bool IsHttpStatusCodeRetryable(HttpStatusCode statusCode, bool hasRetryDelay)
+    {
+        switch (statusCode)
+        {
+#if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
+            case HttpStatusCode.TooManyRequests:
+#else
+            case (HttpStatusCode)429:
```

Comment:
Is this a valid case for `net462` or `netstandard2.0`?

This status code in not defined in `HttpStatusCode` enum for the above targets:
- https://learn.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.6.2
- https://learn.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netstandard-2.0

Reply:
Yes, a 429 HTTP status code is valid irrespective of version of .NET. It's just that the HttpStatusCode enum did not include an explicit value for it in older versions.

Reply:
I was actually concerned about the `enum` casting for an undefined value but looks like it would be fine.